### DISCUSSION TITLE
cant open pdf file after report generation, the file is still in use

### DISCUSSION THREAD

**Discussion by houssamfarag** [acesseonline/pyreportjasper#98] [Q&A] [answered]
def mysql_to_pdf(self):

        CONNECTOR_DIR = os.path.abspath(os.path.dirname(__file__))
        RESOURCES_DIR = os.path.join(os.path.dirname(__file__), "reports")
        REPORTS_DIR = os.path.join(os.path.dirname(__file__), "reports")
        input_file = os.path.join(REPORTS_DIR, 'invoice.jrxml')
        output_file = os.path.join(REPORTS_DIR, 'report')
        pyreportjasper = PyReportJasper()
        pyreportjasper.config(
          input_file,
          output_file,
          output_formats=["pdf"],
          db_connection={
            'driver': 'mysql',
            'username': 'root',
            'password': 'Mysqlhoss9009',
            'host': 'localhost',
            'database': 'sql_decisions',
            'port': '3306',
            'jdbc_driver': 'com.mysql.cj.jdbc.Driver',
            'jdbc_dir': CONNECTOR_DIR
          },
          resource=RESOURCES_DIR,
          parameters={'Parameter1': "8419"}
        )
        pyreportjasper.process_report()
        self.open_report(output_file)

    def open_report(self, output_file):
        os.startfile(output_file + '.pdf')

**Comment by jadsonbr** [comment]
This can be several things among these things is antivirus scanning the file among so many others.

In relation to the library, did you have any problems in the log?
Are you manually checking the final file exists from the informed directory?
Did opening the report manually with your operating system's PDF viewer have any errors?

Tell us a little more about your experience.

**Reply by houssamfarag** [reply]
i can open in my os but i have to close the application 
looks like it is preventing access to file 
i tried to create 2 files in 2 different functions and opening the first file was fine 
so it is locking the access to the last file created

**Reply by houssamfarag** [reply]
i am using pyside2 btw

**Reply by houssamfarag** [reply]
starting the old files worked with pyside2 also .. the file size keeps 0 bytes untill i create another or close the application

**Reply by jadsonbr** [reply]
There must be something inside your application or operating system that is preventing or loading the file and is not releasing it from memory.

I ran some tests here and generated 5 files successfully and none were stuck.

**Reply by houssamfarag** [reply]
i tried with xlsx and it worked but pdf i need to create another to open the old or close the application

**Reply by jadsonbr** [reply]
As I mentioned earlier something is blocking your file. And I believe it has nothing to do with the library.

**Reply by houssamfarag** [reply]
is there something wrong with this as it starts the jvm
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'org.apache.logging.log4j.simplelog.StatusLogger.level' to TRACE to show Log4j2 internal initialization logging.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by br.com.acesseonline.classpath.ApplicationClasspath (file:/C:/Users/h/PycharmProjects/Alnafees_Decisions_Testing/venv/Lib/site-packages/pyreportjasper/libs/acesseonline-1.0.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of br.com.acesseonline.classpath.ApplicationClasspath
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

**Reply by jadsonbr** [reply]
This is normal.
The ERROR tag there as it is described is because there is no log file of the informed log4j and the WARNING are informative of the use of our class br.com.acesseonline.classpath.ApplicationClasspath which is necessary to add other jars at run time.

You can see that in any execution of the library you will get this output at the beginning of the terminal.

**Reply by jadsonbr** [reply]
![62](https://user-images.githubusercontent.com/10354607/114234449-2790c100-9955-11eb-903c-351a1eb3eeb1.PNG)

**Comment by lypch** [comment, marked as answer]
I encountered the issue as well and fixed it in my local version. and summitted an issue, please check it in the following link, which included the solution. 
https://github.com/acesseonline/pyreportjasper/issues/121#issue-1131461551

**Reply by houssamfarag** [reply]
I will be testing today
Thanks
